aspnet/2-structured-data/Models/DatastoreBookStore.cs

```diff
@@ -132,112 +129,81 @@ public DatastoreBookStore(string projectId)
-                    DatastoreService.Scope.UserinfoEmail,
-                });
-            }
-            // Create our connection to datastore.
-            _datastore = new DatastoreService(new Google.Apis.Services
-                .BaseClientService.Initializer()
-            {
-                HttpClientInitializer = credentials,
-                ApplicationName = "Bookshelf.NET-Step2"
-            });
-        }
-
-        /// <summary>
-        /// A convenience function which commits a mutation to datastore.
-        /// Use this function to avoid a lot of boilerplate.
-        /// </summary>
-        /// <param name="mutation">The change to make to datastore.</param>
-        /// <returns>The result of commiting the change.</returns>
-        // [START commitmutation]
-        private CommitResponse CommitMutation(Mutation mutation)
-        {
-            var commitRequest = new CommitRequest()
-            {
-                Mode = "NON_TRANSACTIONAL",
-                Mutation = mutation
-            };
-            return _datastore.Datasets.Commit(commitRequest, _projectId)
-                .Execute();
+            _datastore = DatastoreClient.Create();
+            // I like this better.
+            _foo = DatastoreFoo.Create(_projectId);
         }
-        // [END commitmutation]
 
         // [START create]
         public void Create(Book book)
         {
-            var result = CommitMutation(new Mutation()
-            {
-                InsertAutoId = new Entity[] { book.ToEntity() }
-            });
-            book.Id = result.MutationResult.InsertAutoIdKeys.First().Path.First().Id.Value;
+            // P0 Calling .ToInsert() is very weird.
+            // Having two very different ways to insert, update, etc. depending on whether or not
+            // I'm in a transaction is annoying.  I want one *interface* to do it.
+            // How about a NullTransaction where all operations are immediately committed, and the
+            // final .Commit() is a no-op?  Or, make DatastoreFoo and Transaction implement the
+            // same interface?
+            CommitResponse response = _datastore.Commit(_projectId, Mode.NonTransactional, new[] { book.ToEntity().ToInsert() });
+            Key key = response.MutationResults[0].Key;
+            book.Id = key.Path.First().Id;
         }
         // [END create]
 
         public void Delete(long id)
         {
-            CommitMutation(new Mutation()
-            {
-                Delete = new Key[] { id.ToKey() }
-            });
+            // Pretty good.
+            _foo.Delete(id.ToKey());
+            var trans = _foo.BeginTransaction();
+            trans.Delete(id.ToKey());
+            trans.Commit();
         }
 
         // [START list]
         public BookList List(int pageSize, string nextPageToken)
         {
             var query = new Query()
             {
+                // Why do see a squiggly red line under "Book", but not in your snippet?

```

Comment:
You were probably look at the snippets in currently-merged code, rather than the new version - this would now be

```
var query = new Query("Book")
{
     Limit = pageSize
}
```